File: typescript/easy/minOperations.ts
function minOperations(nums: number[], k: number): number {
  let result = 0;
  for (const num of nums) {
    result = result ^ num;
  }
  result = result ^ k;
  let rs = 0;
  while (result > 0) {
    if ((result & 1) == 1) {
      rs++;
    }
    result = result >> 1;
  }
  return rs;
}

File: rust/src/easy/min_operations.rs
use std::cmp::max;

pub fn min_operations(nums: Vec<i32>, k: i32) -> i32 {
    let sum: i32 = nums.iter().sum();
    let a = max(1, 2);
    sum % k
}

#[cfg(test)]
mod tests {
    use super::*;
    #[test]
    fn it_works() {
        assert_eq!(min_operations(vec![1, 2, -2, -3, 4], 2), 3);
    }
}
